### ISSUE TITLE
autoIncrement not working - PhoneGap - Android 4.0

### ISSUE THREAD

**Issue by bugwelle** [indexeddbshim/IndexedDBShim#67]
Hello,
first: thanks a lot for this awesome script :+1: 

But I have a problem: I use `IndexedDBShim` with `db.js` but it seems that `autoIncrement` does not work,...

the schema for db.js looks like this:

```
settings: {
    key: { keyPath: 'id', autoIncrement: true },
    indexes: { id: { unique: true } }
},...
```

Whenever I add something to `settings`, `id` is undefined. In Chrome, `id` is defined... :/

My configuration:
- Android 4.0
- newest `db.js` and `IndexedDBShim`
- PhoneGap 2.9

Before I forget. I add both files to my HTML file, AFTER the `deviceready` event is fired, so the PhoneGap version of WebSQL is used.

Greetings from Germany,
Andre :)

**Comment by torto** [comment]
happens to me too, any solutions?

**Comment by torto** [comment]
My solution: 

I'm using the lib and added db.js on line 345 db.js the following code, but it is static and generates an id, which for my situation resolves.

``` javascript
if(cursor != undefined){
                    if(cursor.value.id === undefined){
                    cursor.value.id = cursor.key;
                    }
                }
```

**Comment by DickvdBrink** [comment]
@tortoyoyo, thanks for the solution. 
I now know it is the same as this issue #108. Hope I have some time to work on it soon! :)

**Comment by icodeforlove** [comment]
@tortoyoyo your solution fixes the ID issue, you should submit a pull request. This seems like a pretty basic use-case.

**Comment by JamesMessinger** [comment]
This should be fixed now (as of version 1.0.5).  Please give it a try and let us know if you find any problems.  Thanks!
